src/commands/classified/external.rs

```diff
@@ -143,62 +133,41 @@ pub(crate) async fn run_external_command(
     }
 
     process = process.cwd(context.shell_manager.path());
-
     trace!(target: "nu::run::external", "cwd = {:?}", context.shell_manager.path());
 
-    let mut process = match stream_next {
-        StreamNext::Last => process,
-        StreamNext::External | StreamNext::Internal => {
-            process.stdout(subprocess::Redirection::Pipe)
-        }
-    };
-
-    trace!(target: "nu::run::external", "set up stdout pipe");
-
-    if let Some(stdin) = stdin {
-        process = process.stdin(stdin);
+    if !is_last {
+        process = process.stdout(subprocess::Redirection::Pipe);
+        trace!(target: "nu::run::external", "set up stdout pipe");
     }
 
-    trace!(target: "nu::run::external", "set up stdin pipe");
     trace!(target: "nu::run::external", "built process {:?}", process);
 
     let popen = process.popen();
 
-    trace!(target: "nu::run::external", "next = {:?}", stream_next);
-
     let name_tag = command.name_tag.clone();
     if let Ok(mut popen) = popen {
-        match stream_next {
-            StreamNext::Last => {
-                let _ = popen.detach();
-                loop {
-                    match popen.poll() {
-                        None => {
-                            let _ = std::thread::sleep(std::time::Duration::new(0, 100000000));
-                        }
-                        _ => {
-                            let _ = popen.terminate();
-                            break;
-                        }
+        if is_last {
```

Comment:
If we could eliminate `is_last`, this would be a large piece of code we could probably rid ourselves of.